ibm/data_source_ibm_is_instance_template.go

```diff
@@ -0,0 +1,629 @@
+// Copyright IBM Corp. 2017, 2021 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package ibm
+
+import (
+	"context"
+
+	"github.com/IBM/vpc-go-sdk/vpcv1"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	isInstanceTemplateHref                 = "href"
+	isInstanceTemplateCrn                  = "crn"
+	isInstanceTemplateLimit                = "limit"
+	isInstanceTemplateNext                 = "next"
+	isInstanceTemplateTotalCount           = "total_count"
+	isInstanceTemplatePortSpeed            = "port_speed"
+	isInstanceTemplatePortType             = "type"
+	isInstanceTemplatePortValue            = "value"
+	isInstanceTemplateDeleteVol            = "delete_volume_on_instance_delete"
+	isInstanceTemplateVol                  = "volume"
+	isInstanceTemplateMemory               = "memory"
+	isInstanceTemplateMemoryValue          = "value"
+	isInstanceTemplateMemoryType           = "type"
+	isInstanceTemplateMemoryValues         = "values"
+	isInstanceTemplateMemoryDefault        = "default"
+	isInstanceTemplateMemoryMin            = "min"
+	isInstanceTemplateMemoryMax            = "max"
+	isInstanceTemplateMemoryStep           = "step"
+	isInstanceTemplateSocketCount          = "socket_count"
+	isInstanceTemplateSocketValue          = "value"
+	isInstanceTemplateSocketType           = "type"
+	isInstanceTemplateSocketValues         = "values"
+	isInstanceTemplateSocketDefault        = "default"
+	isInstanceTemplateSocketMin            = "min"
+	isInstanceTemplateSocketMax            = "max"
+	isInstanceTemplateSocketStep           = "step"
+	isInstanceTemplateVcpuArch             = "vcpu_architecture"
+	isInstanceTemplateVcpuArchType         = "type"
+	isInstanceTemplateVcpuArchValue        = "value"
+	isInstanceTemplateVcpuCount            = "vcpu_count"
+	isInstanceTemplateVcpuCountValue       = "value"
+	isInstanceTemplateVcpuCountType        = "type"
+	isInstanceTemplateVcpuCountValues      = "values"
+	isInstanceTemplateVcpuCountDefault     = "default"
+	isInstanceTemplateVcpuCountMin         = "min"
+	isInstanceTemplateVcpuCountMax         = "max"
+	isInstanceTemplateVcpuCountStep        = "step"
+	isInstanceTemplateStart                = "start"
+	isInstanceTemplateVersion              = "version"
+	isInstanceTemplateBootVolumeAttachment = "boot_volume_attachment"
+)
+
+func dataSourceIBMISInstanceTemplate() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceIBMISInstanceTemplateRead,
+		Schema: map[string]*schema.Schema{
+			"identifier": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Computed:      true,
```

Comment:
Instead of conflictsWith use ExactlyOneOf...We have anoher datasource which list all tempaltes right?